### DISCUSSION TITLE
How does Leaflet handle rendering with different vector data at different zoom?

### DISCUSSION THREAD

**Discussion by xiaxiangfeng** [Leaflet/Leaflet#9276] [Q&A]
For example, when rendering buildings, when zoom changes from 5 to 6, are the rendering objects of zoom 5 deleted and then rendered at zoom 6? Will this cause flickering? How does Leaflet handle preventing flickering?

**Comment by IvanSanchez** [comment]
Leaflet doesn't do this in the core.

If you look at something like https://github.com/Leaflet/Leaflet.VectorGrid/ , which uses the `L.GridLayer` logic, then the answer is "it follows the `L.GridLayer` pruning algorithm. A tile gets pruned (i.e. removed) whenever all parent *and* children tiles have been loaded.
